components/script/dom/xmlhttprequest.rs

```diff
@@ -1068,6 +1067,7 @@ impl XMLHttpRequest {
             _ => { return None; }
         }
         temp_doc.set_encoding_name(DOMString::from(charset.name()));
+        self.response_xml.set(Some(temp_doc.r()));
```

Comment:
Btw, just implementing step 1 is good enough; I'll file an issue about implementing the rest of the steps correctly once this PR gets merged.